core/ledger/ledgermgmt/ledger_mgmt_test.go

```diff
@@ -78,6 +84,88 @@ func TestLedgerMgmt(t *testing.T) {
 	ledgerMgr.Close()
 }
 
+// TestCreateLedgerFromSnapshot first creates a ledger using a genesis block and generates a snapshot.
+// After it, it tests creating ledger from the snapshot.
+func TestCreateLedgerFromSnapshot(t *testing.T) {
+	testDir, err := ioutil.TempDir("", "createledgerfromsnapshot")
+	require.NoError(t, err)
+	initializer, err := constructDefaultInitializer(testDir)
+	require.NoError(t, err)
+	ledgerMgr := NewLedgerMgr(initializer)
+	defer func() {
+		os.RemoveAll(testDir)
+	}()
+
+	channelID := "testcreatefromsnapshot"
+	bg, gb := testutil.NewBlockGenerator(t, channelID, false)
+	l, err := ledgerMgr.CreateLedger(channelID, gb)
+	require.NoError(t, err)
+
+	// create a block on the ledger
+	txid := util.GenerateUUID()
+	simulator, err := l.NewTxSimulator(txid)
+	require.NoError(t, err)
+	require.NoError(t, simulator.SetState("ns1", "key1", []byte("value1")))
+	simulator.Done()
+	simRes, err := simulator.GetTxSimulationResults()
+	require.NoError(t, err)
+	pubSimBytes, err := simRes.GetPubSimulationBytes()
+	require.NoError(t, err)
+	block1 := bg.NextBlockWithTxid([][]byte{pubSimBytes}, []string{txid})
+	require.NoError(t, l.CommitLegacy(&ledger.BlockAndPvtData{Block: block1}, &ledger.CommitOptions{}))
+
+	bcInfo, err := l.GetBlockchainInfo()
+	require.NoError(t, err)
+	block1Hash := protoutil.BlockHeaderHash(block1.Header)
+	gbHash := protoutil.BlockHeaderHash(gb.Header)
+	require.Equal(t, &common.BlockchainInfo{
+		Height: 2, CurrentBlockHash: block1Hash, PreviousBlockHash: gbHash,
+	}, bcInfo)
+
+	// generate a snapshot
+	require.NoError(t, l.SubmitSnapshotRequest(2))
+	snapshotDir := kvledger.SnapshotDirForLedgerHeight(initializer.Config.SnapshotsConfig.RootDir, channelID, 2)
+	snapshotExists := func() bool {
+		stat, err := os.Stat(snapshotDir)
+		if err != nil {
+			return false
+		}
+		return stat != nil
+	}
+	require.Eventually(t, snapshotExists, 30*time.Second, 100*time.Millisecond)
+
+	// sleep to ensure generateSnapshot func is returned due to the time gap between shapshotDir created and kvledger.generateSnapshot returned.
+	// Then close ledgerMgr.
+	time.Sleep(100 * time.Millisecond)
+	ledgerMgr.Close()
+
+	// re-create the ledger from snapshot under the same rootdir should return error because the ledger already exists
+	ledgerMgr = NewLedgerMgr(initializer)
+	_, _, err = ledgerMgr.CreateLedgerFromSnapshot(snapshotDir)
+	require.EqualError(t, err, "error while creating ledger id: ledger [testcreatefromsnapshot] already exists with state [ACTIVE]")
+
+	// re-create the ledger from snapshot under a different root dir should work because the idStore is empty
+	testDir2, err := ioutil.TempDir("", "createledgerfromsnapshot2")
+	require.NoError(t, err)
+	initializer2, err := constructDefaultInitializer(testDir2)
+	require.NoError(t, err)
+
+	ledgerMgr2 := NewLedgerMgr(initializer2)
+	defer func() {
+		os.RemoveAll(testDir2)
+		ledgerMgr2.Close()
+	}()
```

Comment:
Does not matter much in the test code, but the sequence looks odd.. can you reverse these statements.